File: src/category_strings/q8_longest_substring_no_repeats/LongestSubstringNoRepeats.js
const LogUtil = require('../../utils/LogUtil.js');

/**
 * Determine the length of the longest substring without any repeating characters.
 * See README for details
 * @param {string} s -input string with any ASCII characters 
 * @return {int} length of longest substring without any repeating characters
 */
var longestStringNoRepeats = function(s) {
    // Write your code here
    return 10;
}

var test = function() {
    // Change these values to manually test your program
    let strInput = "aabcc";
    let acceptedResponse = 3;
    let actualResponse = longestStringNoRepeats(strInput);
    new LogUtil().logResultsInt("8: Longest Substring No Repeats", acceptedResponse, actualResponse);
}

module.exports = test;

File: src/category_strings/q8_longest_substring_no_repeats/LongestSubstringNoRepeats.py
from utils.LogUtil import LogUtil;

class LongestSubstringNoRepeats:
    
    # Determine the length of the longest substring without any repeating characters.
    # See README for details
    # @param s - input string with any ASCII characters 
    # @return - length of longest substring without any repeating characters
    def longest_substring_no_repeats(self, s: str) -> str:
        # Write your code here
        return 10;

    def test(self):
        # Change these values to manually test your program
        str_input = "aabcc";
        accepted_response = 3
        actual_response = self.longest_substring_no_repeats(str_input);
        LogUtil().log_results_int("8: Longest Substring No Repeats", accepted_response, actual_response);